src/os/bluestore/bluestore_types.cc

```diff
@@ -17,6 +17,32 @@
 #include "common/Checksummer.h"
 #include "include/stringify.h"
 
+void ExtentList::add_extents(int64_t start, int64_t count) {
+  AllocExtent *last_extent = NULL;
+  bool can_merge = false;
+
+  if (m_num_extents > 0) {
+    last_extent = &((*m_extents)[m_num_extents - 1]);
+    uint64_t last_offset = (last_extent->offset + last_extent->length) / 
+			m_block_size; 
+    uint32_t last_length = last_extent->length / m_block_size; 
+    int64_t max_blocks = m_max_alloc_size / m_block_size;
+    if ((last_offset == (uint64_t) start) &&
+        (!max_blocks || (last_length + count) <= max_blocks)) {
+      can_merge = true;
+    }
+  }
+
+  if (can_merge) {
+    last_extent->length += (count * m_block_size);
+  } else {
+    (*m_extents)[m_num_extents].offset = start * m_block_size;
+    (*m_extents)[m_num_extents].length = count * m_block_size;
+    m_num_extents++;
+  }
+  assert((int64_t) m_extents->size() >= m_num_extents);
```

Comment:
Shouldn't we have assert( m_extents->size() > m_num_extents ) at the function entrance instead?